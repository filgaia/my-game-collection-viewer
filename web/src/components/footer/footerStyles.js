const footerStyles = (theme) => ({
    footer: {
        backgroundColor: theme.palette.background.paper,
        padding: theme.spacing.unit * 6,
    }
});

export default footerStyles;
