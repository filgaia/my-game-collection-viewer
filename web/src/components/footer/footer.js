// @vendors
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// @material
import { Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
// @styles
import footerStyles from './footerStyles';

class Footer extends Component {
    render() {
        const { classes } = this.props;

        return (
            <footer className={classes.footer}>
                <Typography variant="h6" align="center" gutterBottom>
                    Filgaia developments
                </Typography>
                <Typography variant="subtitle1" align="center" color="textSecondary" component="p">
                    Alone in the fields
                </Typography>
            </footer>
        );
    }
}

Footer.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(footerStyles)(Footer);
