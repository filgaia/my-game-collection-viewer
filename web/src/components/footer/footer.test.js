import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Footer } from './footer';

describe('<Footer />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Footer classes={{}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders title', () => {
    render(<Footer classes={{}} />);
    const linkElement = screen.getByText(/Filgaia developments/i);
    expect(linkElement).toBeInTheDocument();
  });
});
