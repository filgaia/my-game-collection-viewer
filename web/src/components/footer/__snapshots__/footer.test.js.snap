// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
<footer>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter MuiTypography-gutterBottom css-tcrtuv-MuiTypography-root"
  >
    Filgaia developments
  </h6>
  <p
    className="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignCenter css-1xatcrf-MuiTypography-root"
  >
    Alone in the fields
  </p>
</footer>
`;
